runtime-complexity/memoization.md

```diff
@@ -1,152 +1,291 @@
-are pure functions. On slide 6 reveal the results. State why the other functions are impure functions. 
 
-The Fibonacci Sequence problem is a well known example of recursion and provides a great way to see many different options for solving the problem and determining the best runtime. The Fibonacci Sequence is created so that each number is the sum of the two preceding numbers. 
+are pure functions. On slide 6 reveal the results. State why the other functions are impure functions.
+
+  
+
+The Fibonacci Sequence problem is a well known example of recursion and provides a great way to see many different options for solving the problem and determining the best runtime. The Fibonacci Sequence is created so that each number is the sum of the two preceding numbers.
+
 [Fibonacci Number Image](https://en.wikipedia.org/wiki/Fibonacci_number#/media/File:34*21-FibonacciBlocks.png)
+
 Pause on slide 9 and write the code to implement the Fibonacci Sequence with a while loop.
 
-View the solution on slide 10 and compare to your own code. 
+  
+
+View the solution on slide 10 and compare to your own code.
+
+  
 
 Next try to implement the Fibonacci Sequence using recursion. Really do this before looking at the solution on slide 12!
 
-Notice the difference in Time Complexity between using recursion and a while loop. 
+  
+
+Notice the difference in Time Complexity between using recursion and a while loop.
+
+  
+
+Finally look at the example code of the Fibonacci Sequence using memoization on slide 13 and again notice the Time Complexity and Space Complexity.
+
+  
+
+Take a look at the comparison chart showing runtime complexity and space complexity for different sample size. what is your biggest take away from slides 17 and 18?
 
-Finally look at the example code of the Fibonacci Sequence using memoization on slide 13 and again notice the Time Complexity and Space Complexity. 
+  
 
-Take a look at the comparison chart showing runtime complexity and space complexity for different sample size. what is your biggest take away from slides 17 and 18? 
+Use the resources on the next few slides to solidify your understanding. Go back and review any of the vocabulary thta you were unsure about at first after you watch the video and go to the links provided.
 
-Use the resources on the next few slides to solidify your understanding. Go back and review any of the vocabulary thta you were unsure about at first after you watch the video and go to the links provided. 
+  
 
 ### Common Mistakes / Misconceptions
 
+  
+
 List things that apprentices might not realize, might assume at first, or should avoid.
 
+  
+
 - Remember: It’s best to implement memoization on functions that are pure and have repetitive calculations.
 
+  
+  
 
 ### Guided Practice
 
+  
+
 Work step by step to change a recursion of Fibonacci Sequence to a memoization version:
 
+  
+
 Recursion
+
 ```
+
 function fibonacci(num) {
+
```

Comment:
Please look at these in the markdown preview and add appropriate indentation.*

Reply:
Almost, but not right yet. Also, are you familiar with JS syntax? You need brackets with if statements { } .